scanner/decisions/0005-scannerv4-read-cvss-data.md

```diff
@@ -0,0 +1,36 @@
+
+The NVD JSON feed offers bundled data on a yearly basis, spanning from 2002 to the present year. A benefit is that only two requests are needed in the workflow for each year (one for the meta file and one for the data bundle). However, a limitation is the absence of filtering capabilities, such as by CVSS score or modification date. In contrast, the NVD API allows data retrieval through specific HTTP URL parameters, such as publish date, modified date, or CVSS score, enabling more precise data acquisition.
+
+## Decision
+
+The updated NVD CVSS GitHub Workflow (CI) has transitioned to sourcing data from the NVD CVE API, rather than the NVD Json feeds, ensuring the timely update of the NVD CVSS data bundle in Google Storage. To fetch data, the workflow downloads based on both the start index and CVSS v3 severity. For instance, a command such as curl "https://services.nvd.nist.gov/rest/json/cves/2.0?cvssV3Severity=CRITICAL&startIndex=0" > critical-1.json would be used. For every severity level, the workflow iteratively exhaust the startIndex to capture all associated data. And there are four severity levels in total. After all json files downloaded, they will be compress to one data bundle as one single zip file.
+
+We've chosen to base our API requests on CVSS V3 severity to ensure all data integrated into Claircore possesses valid CVSS V3 metrics. Consequently, retrieving data without CVSS V3 information becomes irrelevant.
+
+The CVSS Updater in Central will be modified to download zip file (generated by CI mentioned above) in the Google bucket. This eliminates the need for JSON parsing in the updater. ScannerV4 will pull the zip bundle from Central and populate the Matcher DB with the enrichment data.
+
+The CVSS Updater in Central will attempt to download the data bundle from the Google bucket a maximum of five times. Should all five attempts be unsuccessful, the updater will pause its efforts and initiate a fresh download attempt after a 4-hour interval. After five unsuccessful download attempts, a failure notification will be swiftly sent to the Scanner on-call Slack channel.
+
+## Consequences
+
+One drawback of the NVD CVE API is its rate limit: only 5 requests are allowed per 30 seconds without an API Key. This restriction necessitates that our workflow fetch data at a slower pace. Additionally, transitioning from the NVD JSON feed to the NVD CVE API brings in pagination limitations. For example, when fetching data with a "low" severity, the API provides a maximum of 2000 CVEs per response. This means multiple requests are required to retrieve the complete dataset for "low" severity. The result is an increased number of smaller JSON files. Consequently, we need to compress all JSON files into a single zip archive.
```

Comment:
What's stopping us from requesting an API key? Seems like an easy consequence to get past.

Reply:
We can get an API key but I just didn't want to include something undetermined in the ADR.